examples/custom-node/src/evm/config.rs

```diff
@@ -33,30 +34,35 @@ impl ConfigureEvm for CustomEvmConfig {
         &self.block_assembler
     }
 
-    fn evm_env(&self, header: &Header) -> EvmEnv<OpSpecId> {
+    fn evm_env(&self, header: &CustomHeader) -> EvmEnv<OpSpecId> {
         self.inner.evm_env(header)
     }
 
     fn next_evm_env(
         &self,
-        parent: &Header,
+        parent: &CustomHeader,
         attributes: &OpNextBlockEnvAttributes,
     ) -> Result<EvmEnv<OpSpecId>, Self::Error> {
         self.inner.next_evm_env(parent, attributes)
     }
 
-    fn context_for_block(
-        &self,
-        block: &SealedBlock<Block<OpTransactionSigned>>,
-    ) -> OpBlockExecutionCtx {
-        self.inner.context_for_block(block)
+    fn context_for_block(&self, block: &SealedBlock<Block>) -> OpBlockExecutionCtx {
+        OpBlockExecutionCtx {
+            parent_hash: block.header().parent_hash(),
+            parent_beacon_block_root: block.header().parent_beacon_block_root(),
+            extra_data: block.header().extra_data().clone(),
+        }
```

Comment:
~~should we do a `From<T:BlockHeader>` impl for OpBlockExecutionCtx ?~~

should we make this a native constructor of OpBlockExecutionCtx

```rust
 fn for_header<T:BlockHeader>(
```